### PR TITLE
SPUI-280/188 > Wallet screen

### PR DESCRIPTION
**Pull request by VovaMedvediev**

### CODE REVIEW

File: lib/app/screens/wallet/notifier/operation_history_notifier.dart [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+import '../../../../service/services/operation_history/model/operation_history_response_model.dart';
+import '../../../../shared/logging/levels.dart';
+import '../../../../shared/providers/service_providers.dart';
+
+class OperationHistoryNotifier
+    extends StateNotifier<List<OperationHistoryItem>> {
+  OperationHistoryNotifier({
+    required this.read,
+  }) : super(
+          [],
```
Comment (eli1stark) [inline lib/app/screens/wallet/notifier/operation_history_notifier.dart:14]: remove comma

File: lib/app/screens/wallet/notifier/operation_history_notifier.dart [hunk 2]
```diff
@@ -0,0 +1,61 @@
...
+      final operationHistory = await _requestOperationHistory(assetId);
+
+      updateOperationHistory(operationHistory.operationHistory);
+    } catch (e) {
+      _logger.log(stateFlow, 'operationHistory', e);
+    }
+  }
+
+  void updateOperationHistory(List<OperationHistoryItem> items) {
+    state = state + items;
```
Comment (eli1stark) [inline lib/app/screens/wallet/notifier/operation_history_notifier.dart:48]: add log

File: lib/app/screens/wallet/provider/hidden_state_provider.dart [hunk 3]
```diff
@@ -0,0 +1,5 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+
+final hiddenStatePod = StateProvider.autoDispose<bool>(
```
Comment (eli1stark) [inline lib/app/screens/wallet/provider/hidden_state_provider.dart:3]: `hiddenStpod`, and maybe somehow clarify the name?

File: lib/app/screens/wallet/view/components/action_block.dart [hunk 4]
```diff
@@ -0,0 +1,23 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+
+import '../../../../../shared/components/buttons/app_buton_white.dart';
+import '../../../market_details/view/components/balance_block/components/balance_frame.dart';
+
+class ActionBlock extends StatelessWidget {
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/action_block.dart:7]: rename to `ActionButton`

File: lib/app/screens/wallet/view/components/wallet_app_bar/empty_wallet_app_bar.dart [hunk 5]
```diff
@@ -0,0 +1,58 @@
...
+                icon: Icons.arrow_back,
+                onTap: () => Navigator.of(context).pop(),
+              ),
+              Expanded(
+                child: HeaderText(
+                  text: '$assetName wallet',
+                  textAlign: TextAlign.center,
+                ),
+              ),
+              SizedBox(
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_app_bar/empty_wallet_app_bar.dart:36]: Refactor to `SpaceW32`

File: lib/app/screens/wallet/view/components/wallet_app_bar/wallet_app_bar.dart [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+          ),
+          Expanded(
+            child: HeaderText(
+              text: containsSingleElement(itemsWithBalance)
+                  ? 'Wallet'
+                  : 'Wallets',
+              textAlign: TextAlign.center,
+            ),
+          ),
+          SizedBox(
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_app_bar/wallet_app_bar.dart:37]: Refactor to `SpaceW32`

File: lib/app/screens/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart [hunk 7]
```diff
@@ -0,0 +1,93 @@
...
+    final hidden = useProvider(hiddenStatePod);
+
+    return Expanded(
+      child: Container(
+        height: 0.25.sh,
+        margin: EdgeInsets.symmetric(
+          horizontal: 24.w,
+        ),
+        padding: EdgeInsets.all(
+          20.r,
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart:35]: remove comma

File: lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart [hunk 8]
```diff
@@ -0,0 +1,130 @@
...
+      child: Column(
+        children: [
+          Row(
+            children: [
+              Icon(
+                _icon(transactionListItem.operationType),
+                size: 20.r,
+              ),
+              const SpaceW10(),
+              Text(
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart:35]: refactor

File: lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart [hunk 9]
```diff
@@ -0,0 +1,130 @@
...
+                _operationName(transactionListItem.operationType),
+                style: TextStyle(
+                  fontWeight: FontWeight.bold,
+                  fontSize: 18.sp,
+                ),
+              ),
+              const Spacer(),
+              Text(
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart:43]: refactor

File: lib/app/screens/wallet/view/components/wallet_body/wallets.dart [hunk 10]
```diff
@@ -0,0 +1,77 @@
...
+                onPageChanged: (index) {
+                  transactionHistoryN.initOperationHistory(
+                    itemsWithBalance[index].associateAsset,
+                  );
+                },
+                children: [
+                  for (final item in itemsWithBalance)
+                    WalletBody(
+                      assetId: item.associateAsset,
+                      currentPage: (_pageController.hasClients
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_body/wallets.dart:55]: refactor to function

File: lib/app/screens/wallet/view/components/wallet_body/wallets.dart [hunk 11]
```diff
@@ -0,0 +1,77 @@
...
+      operationHistoryNotipod.notifier,
+    );
+    final itemsWithBalance = assetsWithBalanceFrom(useProvider(marketItemsPod));
+
+    return transactionsHistoryInit.when(
+      data: (_) {
+        return RefreshIndicator(
+          onRefresh: () => _refresh(
+            context,
+            itemsWithBalance[(_pageController.page ?? 0).toInt()]
```
Comment (eli1stark) [inline lib/app/screens/wallet/view/components/wallet_body/wallets.dart:37]: refactor to function

File: lib/router/view/components/app_frame.dart [hunk 12]
```diff
@@ -3,18 +3,24 @@ import 'package:flutter/material.dart';
 class AppFrame extends StatelessWidget {
```
Comment (eli1stark) [inline lib/router/view/components/app_frame.dart:3]: Refactor to lib/shared/components

**Review by eli1stark — CHANGES_REQUESTED** [review]

File: lib/app/screens/wallet/notifier/operation_history_notifier.dart (hunk 1 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/notifier/operation_history_notifier.dart:14]: removed

File: lib/app/screens/wallet/notifier/operation_history_notifier.dart (hunk 2 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/notifier/operation_history_notifier.dart:48]: added

File: lib/app/screens/wallet/provider/hidden_state_provider.dart (hunk 3 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/provider/hidden_state_provider.dart:3]: renamed both file and variable to walletHiddenStPod

File: lib/app/screens/wallet/view/components/action_block.dart (hunk 4 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/action_block.dart:7]: renamed

File: lib/app/screens/wallet/view/components/wallet_app_bar/empty_wallet_app_bar.dart (hunk 5 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_app_bar/empty_wallet_app_bar.dart:36]: refactored

File: lib/app/screens/wallet/view/components/wallet_app_bar/wallet_app_bar.dart (hunk 6 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_app_bar/wallet_app_bar.dart:37]: refactored

File: lib/app/screens/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart (hunk 7 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_body/components/card_block/components/wallet_card.dart:35]: removed

File: lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart (hunk 8 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart:35]: refactored

File: lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart (hunk 9 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_body/components/transactions_list/transaction_list_item.dart:43]: refactored

File: lib/app/screens/wallet/view/components/wallet_body/wallets.dart (hunk 10 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_body/wallets.dart:55]: refactored

File: lib/app/screens/wallet/view/components/wallet_body/wallets.dart (hunk 11 above)
Comment (VovaMedvediev) [reply lib/app/screens/wallet/view/components/wallet_body/wallets.dart:37]: refactored

File: lib/app/screens/wallet/helper/assets_with_balance_from.dart [hunk 13]
```diff
@@ -0,0 +1,5 @@
+import '../../market/model/market_item_model.dart';
+
+List<MarketItemModel> assetsWithBalanceFrom(List<MarketItemModel> assets) {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/wallet/helper/assets_with_balance_from.dart:3]: comment added

File: lib/app/screens/wallet/helper/market_item_from.dart [hunk 14]
```diff
@@ -0,0 +1,5 @@
+import '../../market/model/market_item_model.dart';
+
+MarketItemModel marketItemFrom(List<MarketItemModel> items, String symbol) {
```
[the remark this answers is not in the document]
Comment (VovaMedvediev) [reply lib/app/screens/wallet/helper/market_item_from.dart:3]: comment added

**Review by eli1stark — APPROVED** [review]

**Pull request merged**